android-pdf-viewer/src/main/java/com/github/barteksc/pdfviewer/annotation/core/pdf/PdfUtil.kt

```diff
@@ -0,0 +1,367 @@
+                                        )
+                                    )
+                                }
+
+                            }
+                        }
+                        val squareAnnotation = Annotation(
+                           AnnotationType.SQUARE.name,
+                            squareAnnotationPoints,
+                            relations = Relations(documentations)
+                        )
+                        annotationsList.add(squareAnnotation)
+
+                        logDebug(TAG, "Annotation is square")
+                        logDebug(TAG, "Annotation $i on page $pageNum - points:")
+                        logDebug(TAG, "squareAnnotationPoints:$squareAnnotationPoints")
+
+                    } else if (subtype == PdfName.CIRCLE) {
+                        // bottom left
+                        val xBottomLeftPoint = llx
+                        val yBottomLeftPoint = lly
+                        val bottomLeftPoint = PointF(xBottomLeftPoint, yBottomLeftPoint)
+
+                        // top right
+                        val xTopRightPoint = urx
+                        val yTopRightPoint = ury
+                        val topRightPoint = PointF(xTopRightPoint, yTopRightPoint)
+
+                        val circleAnnotationPoints = listOf(bottomLeftPoint, topRightPoint)
+                        val circleAnnotation = Annotation("CIRCLE", circleAnnotationPoints)
+                        annotationsList.add(circleAnnotation)
+
+                        logDebug(TAG, "Annotation is circle")
+                        logDebug(TAG, "Annotation $i on page $pageNum - points:")
+                        logDebug(TAG, "circleAnnotationPoints:$circleAnnotationPoints")
+                    } else {
+                        logDebug(TAG, "Annotation is not recognised")
+                    }
+                }
+            }
+        }
+
+        return annotationsList
+    }
+
+    /** Map annotations to shapes,
+     *  using the page height when converting between PDF space and image space*/
+    private fun getShapesFor(
+        pdfAnnotations: List<Annotation>,
+        pageHeight: Int
+    ): List<Shape> {
+        // convert annotation to shape
+        val shapes = pdfAnnotations.map { annotation ->
+            when (annotation.type) {
+                AnnotationType.SQUARE.name -> return@map annotation.toRectangleShape(pageHeight)
+                AnnotationType.CIRCLE.name -> return@map annotation.toCircleShape(pageHeight)
+                else -> throw Exception("Annotation is not recognised")
+            }
+        }
+        return shapes
+    }
+
+    /** Use the passed PDF file path to map the PDF page to PNG image and
+     *  map PDF annotations to image shapes, save the PNG image to the given output directory
+     */
+    @JvmStatic
+    @Throws(IOException::class)
+    private fun convertPdfAnnotationsToPngShapes(
```

Comment:
Please cleanup this from white lines, redundant comments

Reply:
Cleaned up

Reply:
Cleaned up